docs/data.fetch.md

```diff
@@ -0,0 +1,135 @@
+# qgrid filtering interface
+
+Filtering is tightly coupled with ordering, paging and virtualization.
+
+Data fetch process is going to be a pipeline of ordered operations which can be configured by user.
+
+Every operation gets data from previous stage and handles and passes to the next stage.
+If user wants to perform some of this actions on client-side, he modifies pipeline by adding a function to corresponding model.
+
+By default the whole pipeline is triggered when some pipeline-related property is changed (e.g. filters, order etc.).
+User also has an ability to trigger data refresh.
+
+Pipeline requires a function with following signature: `(data, model, next) => { }`
+where `data` is data array from previous stage, 
+`model` is grid model and 
+`next` is function that finishes current stage and passes data to the next one.
+
+## gridService
+
+Along with `qGridModel` `qGridService` must be exported too to encapsulate some operations on model.
+```javascript
+var model = qGridModel(),
+    service = qGridService(model);
+
+service.invalidate();
+```  
+
+## Common cases
+1. All data is filtered and ordered on server-side.
+```javascript
+gridModel
+    .data({
+        pipeline: [
+            (data, model, next) => {
+                $http({
+                    filter: model.filter,
```

Comment:
what is the model? Maybe just pass gridmodel to make all settings available, maybe better to have
context = {
 source: 'filter|manual|sort... why invalidate was triggered'
 model: gridmodel,
 changes: 'what was changes'
}
```javascript
(data, ctx, next) => {
    const dataState = ctx.model.data();
    const filter = ctx.data().filter;
    const selection = ctx.selection().items; //filttration can depened on selection for example
}
```